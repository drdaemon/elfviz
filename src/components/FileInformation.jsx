import React from 'react';

const FileInformation = ({ file }) => {
  return file ? <h5>{file.name} {file.size} bytes</h5> : null;
};

export default FileInformation;
