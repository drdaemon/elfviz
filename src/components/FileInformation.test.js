import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FileInformation from './FileInformation';

describe('FileInformation', () => {
  it('renders without crashing', () => {
    shallow(<FileInformation />);
  });

  it('should match an empty snapshot', () => {
    const wrapper = shallow(<FileInformation />);
    expect(toJson(wrapper)).toBeNull();
  });

  it('should match a snapshot with file info', () => {
    const wrapper = shallow(<FileInformation file={{ name: 'test file', size: 12345 }} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
