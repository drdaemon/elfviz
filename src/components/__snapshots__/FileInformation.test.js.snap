// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileInformation should match a snapshot with file info 1`] = `
<h5>
  test file
   
  12345
   bytes
</h5>
`;
